pkg/agent/controller/packetcapture/packetcapture_controller_linux.go

```diff
@@ -0,0 +1,879 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package packetcapture
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/gopacket/gopacket"
+	"github.com/gopacket/gopacket/layers"
+	"github.com/gopacket/gopacket/pcapgo"
+	"github.com/spf13/afero"
+	"golang.org/x/time/rate"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/client-go/util/workqueue"
+	klog "k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/controller/packetcapture/bpf"
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/util"
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	clientsetversioned "antrea.io/antrea/pkg/client/clientset/versioned"
+	crdinformers "antrea.io/antrea/pkg/client/informers/externalversions/crd/v1alpha1"
+	crdlisters "antrea.io/antrea/pkg/client/listers/crd/v1alpha1"
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+	"antrea.io/antrea/pkg/util/env"
+	"antrea.io/antrea/pkg/util/ftp"
+)
+
+type StorageProtocolType string
+
+const (
+	sftpProtocol StorageProtocolType = "sftp"
+)
+
+const (
+	controllerName               = "AntreaAgentPacketCaptureController"
+	resyncPeriod   time.Duration = 0
+
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+
+	defaultWorkers = 4
+
+	maxWaitingCaptures = 32
+
+	// reason for timeout
+	captureTimeoutReason   = "PacketCapture timeout"
+	defaultTimeoutDuration = time.Second * time.Duration(60)
+
+	captureStatusUpdatePeriod = 10 * time.Second
+
+	// PacketCapture uses a dedicated secret object to store auth info for file server.
+	// #nosec G101
+	fileServerAuthSecretName = "antrea-packetcapture-fileserver-auth"
+
+	// Max packet size for pcap capture.
+	snapshotLen = 65536
+)
+
+var (
+	packetDirectory = getPacketDirectory()
+	defaultFS       = afero.NewOsFs()
+)
+
+func getPacketDirectory() string {
+	return filepath.Join(os.TempDir(), "antrea", "packetcapture", "packets")
+}
```

Comment:
It's unnecessary to wrap it as a function unless it will be called multiple times for a one-line function.

```suggestion
var (
	packetDirectory = filepath.Join(os.TempDir(), "antrea", "packetcapture", "packets")
	defaultFS       = afero.NewOsFs()
)
```